app/BUILD.bazel

```diff
@@ -336,6 +388,70 @@ kt_android_library(
     visibility = ["//visibility:private"],
 )
 
+# Library for all custom data-binding adapters under org/oppia/app/databinding
+android_library(
+    name = "databinding_adapters",
+    custom_package = "org.oppia.app.impl.adapters",
+    srcs = DATABINDING_ADAPTER_FILES,
+    enable_data_binding = 1,
+    manifest = "src/main/DbAdaptersAndroidManifest.xml",
+    deps = [
+        ":dagger",
+        "//model",
+        "//utility",
+        ":view_models",
+        ":resources",
+        artifact("androidx.appcompat:appcompat"),
+        artifact("com.chaos.view:pinview:1.4.3"),
+        artifact("com.github.bumptech.glide:glide"),
+        artifact("com.caverock:androidsvg-aar"),
+        artifact("androidx.lifecycle:lifecycle-livedata-ktx"),
+        artifact("androidx.databinding:databinding-runtime"),
+        artifact("androidx.databinding:databinding-adapters"),
+        artifact("androidx.databinding:databinding-common"),
+        artifact("androidx.annotation:annotation"),
+        artifact("com.google.android.material:material:1.2.0-alpha02"),
+        artifact("androidx.constraintlayout:constraintlayout:1.1.3"),
+        artifact("de.hdodenhof:circleimageview:3.0.1"),
+        artifact("javax.annotation:javax.annotation-api:jar"),
+        artifact("org.jetbrains:annotations:jar"),
+        "@circularimageview//circularimageview:circular_image_view",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+# Library for all custom databinding adapters under org/oppia/app/recyclerview
+android_library(
+    name = "recyclerview_adapters",
+    custom_package = "org.oppia.app.rv",
+    srcs = RECYCLERVIEW_ADAPTER_FILES,
+    enable_data_binding = 1,
+    manifest = "src/main/RvAdaptersAndroidManifest.xml",
+    deps = [
+        ":dagger",
+        "//model",
+        "//utility",
+        ":view_models",
+        artifact("androidx.appcompat:appcompat"),
+        artifact("com.chaos.view:pinview:1.4.3"),
+        artifact("com.github.bumptech.glide:glide"),
+        artifact("com.caverock:androidsvg-aar"),
+        artifact("androidx.lifecycle:lifecycle-livedata-ktx"),
+        artifact("androidx.databinding:databinding-runtime"),
+        artifact("androidx.databinding:databinding-adapters"),
+        artifact("androidx.databinding:databinding-common"),
+        artifact("androidx.annotation:annotation"),
+        artifact("com.google.android.material:material:1.2.0-alpha02"),
+        artifact("androidx.constraintlayout:constraintlayout:1.1.3"),
+        artifact("de.hdodenhof:circleimageview:3.0.1"),
+        artifact("javax.annotation:javax.annotation-api:jar"),
+        artifact("org.jetbrains:annotations:jar"),
+        "@circularimageview//circularimageview:circular_image_view",
+        artifact("androidx.lifecycle:lifecycle-livedata-core:jar:2.0.0-alpha1"),
+    ],
+    visibility = ["//visibility:public"],
```

Comment:
Can we set the visibility per BUILD file (i.e. package defaults) instead of per target? Also, do these need to be public? Can we instead specify exactly which BUILD files need these as dependencies?

Reply:
These do not need to be public - the only library that should be public is `app`. However, if you look at the bottom of the file, where the firebase dependencies are, the package default visibility is stated there. Because some of the firebase rules do not have visibility attributes (at least that you can set) and they need to be public (since other libraries depend on them) -- the defualt visibility of this file needs to be public.

Reply:
Okay got it , thanks!